source/vibe/db/postgresql/package.d

```diff
@@ -116,12 +132,26 @@ class __Conn : dpq2.Connection
             destroy(sock);
         }
 
-        auto event = createFileDescriptorEvent(sock.handle, FileDescriptorEvent.Trigger.any);
+        auto event = createFileDescriptorEvent(sock.handle, FileDescriptorEvent.Trigger.read);
         scope(exit) destroy(event); // Prevents 100% CPU usage
 
         if(!event.wait(timeout))
             throw new PostgresClientTimeoutException(__FILE__, __LINE__);
     }
+    override Socket socket()
```

Comment:
This is copy and paste from dpq2? But why?

Reply:
Not exactly, Socket type is not same here it is different for vibe-core

Reply:
What do you mean? This `Socket` should be from dpq2 package.

Reply:
To be fair I thing we even dond need this.socket, it is only usefull for hidding platform dependant code. I will probably rewrite it a little more

Reply:
No it is not, Socket is std.socket.Socket if vibe-core is not used or DestrucNoCloseSocket if vibe-core is used

Reply:
> Socket is std.socket.Socket if vibe-core is not used or DestrucNoCloseSocket if vibe-core is used

I am about current `master`: currently all Postgres connections infrastructure isn't depends from `vibe` or from `libevent`